cpu/stm32_common/periph/rtc.c

```diff
@@ -221,22 +221,29 @@ void rtc_clear_alarm(void)
     rtc_callback.arg = NULL;
 }
 
+
 void rtc_poweron(void)
 {
 #if defined(CPU_FAM_STM32L1)
     /* Reset RTC domain */
     RCC->CSR |= RCC_CSR_RTCRST;
     RCC->CSR &= ~(RCC_CSR_RTCRST);
 
+#if CLOCK_HAS_LSE
     /* Enable the LSE clock (external 32.768 kHz oscillator) */
     RCC->CSR &= ~(RCC_CSR_LSEON);
     RCC->CSR &= ~(RCC_CSR_LSEBYP);
     RCC->CSR |= RCC_CSR_LSEON;
     while ( (RCC->CSR & RCC_CSR_LSERDY) == 0 ) {}
-
     /* Switch RTC to LSE clock source */
     RCC->CSR &= ~(RCC_CSR_RTCSEL);
     RCC->CSR |= RCC_CSR_RTCSEL_LSE;
+#else
+    /* Enable the LSI clock */
+    RCC->CSR = RCC_CSR_LSION;
+    while (!(RCC->CSR & RCC_CSR_LSIRDY)) {}
```

Comment:
For a **F0** this didn't work. The LSI is off after each reboot but [(RCC->BDCR & RCC_BDCR_RTCEN) == 1](https://github.com/RIOT-OS/RIOT/pull/6504/files#diff-bdada3d6cf063f73ecca7e6bd807935dR67) so rtc_poweron() isn't called. Adding these 2 lines to clk_init() in cpu.c solved it for me